notebooks/ingestion/Loading.ipynb

```diff
@@ -0,0 +1,275 @@
+    "    caption: str\n",
+    "    recipe_url: str\n",
+    "    dish_id: int\n",
+    "    ingredients: List[Ingredient]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create objects of these classes.\n",
+    "\n",
+    "We will provision the data using a json file, prepared from [script](https://github.com/aperture-data/Cookbook/blob/main/scripts/create_nested_json.py)\n",
+    "\n",
+    "Example line from the json file:\n",
+    "\n",
+    ":::note title=\"Sample from the dishes.json\"\n",
+    "```\n",
+    "{\n",
+    "        \"url\": \"https://github.com/aperture-data/Cookbook/blob/0e01a8dca2a995e311508ddbf10a9de13c67bb9b/images/001%20Large.jpeg\",\n",
+    "        \"contributor\": \"gautam\",\n",
+    "        \"dish_name\": \"rajma chawal\",\n",
+    "        \"location\": \"NJ\",\n",
+    "        \"food_tags\": \"Indian\",\n",
+    "        \"caption\": \"Beans with rice\",\n",
+    "        \"constraint_id\": 1,\n",
+    "        \"dish_id\": 1,\n",
+    "        \"recipe_url\": \"https://www.tarladalal.com/rajma-chawal-punjabi-rajma-chawal-4951r\",\n",
+    "        \"ingredients\": [\n",
+    "            {\n",
+    "                \"ingredient_name\": \"rajma\"\n",
+    "            },\n",
+    "            {\n",
+    "                \"ingredient_name\": \"rice\"\n",
+    "            }\n",
+    "        ]\n",
+    "    }\n",
+    "```\n",
+    ":::\n",
+    "\n",
+    "These objects can be passed to a function called generate_add_query which takes care of generating the queries that ApertureDB can execute to persist the objects on the DB."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from aperturedb.Query import generate_add_query\n",
+    "from aperturedb.CommonLibrary import  execute_query, create_connector\n",
+    "import json\n",
+    "\n",
+    "\n",
+    "with open(\"dishes.json\") as ins:\n",
+    "    client = create_connector()\n",
+    "    dishes = json.load(ins)\n",
+    "    for dish in dishes:\n",
+    "\n",
+    "        dish = Dish(**dish)\n",
+    "\n",
+    "        query, blobs, _ = generate_add_query(dish)\n",
+    "        result, response, output_blobs = execute_query(client, query, blobs)\n",
+    "\n",
+    "        if result != 0:\n",
+    "            print(response)\n",
+    "            break\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Ingesting using the CSV Parsers.\n",
+    "\n",
+    "Let's ingest the same data. We shall pre process the csvs to be compatible to aperuredb's input CSV format.\n",
+    "The generated schma for the data will be still the same."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from aperturedb.ImageDataCSV import ImageDataCSV\n",
+    "from aperturedb.EntityDataCSV import EntityDataCSV\n",
```

Comment:
Can we add a line with adb ingest from csv for the particular data type?

Reply:
yes, will do.